podcasts/End of Year/EndOfYearStoriesBuilder.swift

```diff
@@ -22,13 +22,22 @@ class EndOfYearStoriesBuilder {
 
     private let data = EndOfYearStoriesData()
 
-    init(dataManager: DataManager = DataManager.sharedManager) {
+    private let sync: (() -> Void)?
+
+    init(dataManager: DataManager = DataManager.sharedManager, sync: (() -> Void)? = nil) {
         self.dataManager = dataManager
+        self.sync = sync
     }
 
     // Call this method to build the list of stories and the data provider
     func build() async -> ([EndOfYearStory], EndOfYearStoriesData) {
         await withCheckedContinuation { continuation in
+            // Check if the user has the full listening history for this year
+            if !dataManager.isFullListeningHistory() {
+                // TODO: update with the correct endpoint to sync history (eoy-todo)
+                sync?()
```

Comment:
Here we'll be able to use something like `SyncHistoryTask` without much effort.

The Stories view is already prepared for it and it will show a loading indicator as long as this goes on.